File: php/awqat/TokenCache.php
<?php

declare(strict_types=1);

namespace Awqat;

/**
 * Token önbelleği: login/refresh sonrası token'ı diske kalıcı yazar.
 * Sonraki çalıştırmalarda geçerliyse yeniden login ATILMAZ (kota korunur).
 * Biçim Go/JS/Python ile UYUMLU: {"accessToken","refreshToken","expiry" (ISO 8601)}.
 */
class TokenCache
{
    /** @return array{accessToken:string,refreshToken?:string,expiry?:string}|null */
    public static function load(string $path): ?array
    {
        if ($path === '' || !is_file($path)) {
            return null;
        }
        $data = json_decode((string) file_get_contents($path), true);
        if (!is_array($data) || empty($data['accessToken'])) {
            return null;
        }
        return $data;
    }

    /** Token'ı yalnızca sahibin okuyabileceği (0600) dosyaya yazar. Hata yutulur. */
    public static function save(string $path, array $tc): void
    {
        if ($path === '') {
            return;
        }
        $json = json_encode($tc, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES | JSON_UNESCAPED_UNICODE);
        if ($json === false) {
            return;
        }
        if (@file_put_contents($path, $json) !== false) {
            @chmod($path, 0600);
        }
    }
}

File: csharp/AwqatSalah/Awqat/TokenCache.cs
using System.Text.Json;

namespace Awqat;

public sealed class TokenCacheData
{
    public string AccessToken { get; set; } = "";
    public string RefreshToken { get; set; } = "";
    public string Expiry { get; set; } = "";
}

/// <summary>
/// Token önbelleği: login/refresh sonrası token'ı diske kalıcı yazar.
/// Sonraki çalıştırmalarda geçerliyse yeniden login ATILMAZ (kota korunur).
/// Biçim Go/JS/Python/PHP ile UYUMLU: {"accessToken","refreshToken","expiry" (ISO 8601)}.
/// </summary>
public static class TokenCache
{
    private static readonly JsonSerializerOptions Options = new()
    {
        PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
        PropertyNameCaseInsensitive = true,
        WriteIndented = true,
    };

    public static TokenCacheData? Load(string path)
    {
        if (string.IsNullOrEmpty(path) || !File.Exists(path)) return null;
        try
        {
            var tc = JsonSerializer.Deserialize<TokenCacheData>(File.ReadAllText(path), Options);
            return tc is not null && tc.AccessToken.Length > 0 ? tc : null;
        }
        catch
        {
            return null;
        }
    }

    /// <summary>Token'ı yalnızca sahibin okuyabileceği (0600) dosyaya yazar. Hata yutulur.</summary>
    public static void Save(string path, TokenCacheData tc)
    {
        if (string.IsNullOrEmpty(path)) return;
        try
        {
            File.WriteAllText(path, JsonSerializer.Serialize(tc, Options));
            if (!OperatingSystem.IsWindows())
            {
                File.SetUnixFileMode(path, UnixFileMode.UserRead | UnixFileMode.UserWrite);
            }
        }
        catch
        {
            // önbellek kritik değil
        }
    }
}
